tripy/docs/pre0_user_guides/02-compiler.md

```diff
@@ -0,0 +1,191 @@
+# Using the Compiler
+
+```{contents} Table of Contents
+:depth: 3
+```
+
+## Eager Mode vs. Compile Mode
+
+Tripy offers two modes of operation: eager mode and compile mode:
+
+### Eager Mode
+
+Eager mode is designed for development and debugging purposes:
+
+1. **Deferred Compilation**:  In eager mode, compilation is deferred until the first use of a tensor, allowing for easier debugging.
+2. **Fast Compilation**: It utilizes specific settings (`builder_optimization_level`) in the TensorRT compiler to skip most optimizations, resulting in faster compilation times.
+3. **Debugging**: Eager mode is particularly helpful when debugging models that are being developed using Tripy APIs.
+4. **Immediate Feedback**: It allows for quick iterations and immediate feedback, which is beneficial during the model development phase.
+
+### Compile Mode
+
+Compile mode is focused on optimization and deployment:
+
+1. **Full Optimization**: Users can leverage the full power of TensorRT's optimizations to improve model's performance.
+2. **Model Serialization**: Compile mode allows users to serialize the compiled model, enabling deployment and distribution.
+3. **Dynamic Shape Support**: Provides the ability to specify a range of shapes for all inputs, allowing for optimized execution across different input sizes.
+
+## Walk through model compilation and deployment
+
+Let's walk through a simple example of a [GEGLU](https://arxiv.org/abs/2002.05202v1) module defined below:
+
+```py
+# doc: no-eval
+class GEGLU(tp.Module):
+    def __init__(self, dim_in, dim_out):
+        self.proj = tp.Linear(dim_in, dim_out * 2)
+        self.dim_out = dim_out
+    
+    def __call__(self, x):
+        proj = self.proj(x)
+        x, gate = tp.split(proj, 2, proj.rank - 1)
+        return x * tp.gelu(gate)
+```
+
+To naively run `GEGLU` in eager mode:
+
+```py
+# doc: no-eval
```

Comment:
I see all code sections are `# doc: no-eval`, is this intended?

Reply:
Yes, no eval means that the doc rendering does not need to execute the code but it is still tested.

Reply:
Why do we not want it to execute code? IMO it's nice to see the output